httpx/_exceptions.py

```diff
@@ -267,6 +268,22 @@ def __init__(self, message: str, *, request: Request, response: Response) -> Non
         self.request = request
         self.response = response
 
+    def __reduce__(self) -> tuple[typing.Any, ...]:
+        # BaseException has a custom __reduce__ that can't handle subclasses with
+        # required keyword-only arguments.
+        return (
+            functools.partial(
+                self.__class__, request=self.request, response=self.response
+            ),
+            self.args,
+            # In case user code adds attributes to the exception instance.
+            {
+                k: v
+                for k, v in self.__dict__.items()
+                if k not in ("_request", "response")
+            },
```

Comment:
The downside here is that everything in `self.__dict__` must be accepted in `__init__`?